beps/0002-dynamic-frontend-plugins/README.md

```diff
@@ -378,153 +612,366 @@ Example of registry with embedded manifests
+      ContainerPlugin,
+    }
+  },
   pluginMetadata: {
-    // version cna be used from the package.json version field
-    version: '1.0.0',
-    /**
-     * Name can be easily derived from the plugin name
-     * https://github.com/janus-idp/backstage-plugins/blob/87a6b045c7b0f301ebed8b8f99dc1741fa2b044b/packages/cli/src/commands/export-dynamic-plugin/frontend.ts#L40
-    */
-    name: 'backstage.plugin-github-actions',
-    /**
-     * Path to the plugin entry point.
-     * It can default to the same entry point as in regular build.
-     * Plugins can expose multiple modules. We have found that one is sufficient from backstage plugins.
-    */
+    name: pluginName,
+    version: plugin.version || '0.0.0',
     exposedModules: {
-      PluginRoot: './src/index.ts'
-    }
+      // path to the default export of the frontend plugin entry point
+      pluginEntry: './src/index.js',
+    },
   },
 });
+
+
+const config: Configuration = {
+    context: pluginRoot,
+    output: {
+      chunkFilename: `${pluginName}.[contenthash].js`,
+      path: path.resolve(pluginRoot, 'dist'),
+      publicPath: 'auto',
+    },
+    entry: {},
+    resolve: {
+      extensions: ['.js', '.jsx', '.ts', '.tsx'],
+    },
+    plugins: [dynamicPluginPlugin],
+    module: {
+      rules: [
+        {
+          // needs TS config as well
+          test: /\.js$/,
+          exclude: /(node_modules)/,
+          use: {
+            loader: 'swc-loader',
+          },
+        },
+        {
+          test: /\.(png|svg|jpg|jpeg|gif)$/i,
+          type: 'asset/resource',
+        },
+        // additional loaders for CSS if necessary, can use the same as the current backstage shell app
+      ],
+    },
+  };
+  return config;
 ```
 
-The `DynamicRemotePlugin` webpack plugin takes care of the rest, including the manifest generation.
+### Plugin manifest
 
-#### Sample plugin raw webpack configuration
+Plugin manifest is a simple JSON file containing critical metadata to initialize a remote plugin.
 
-The Scalprum config translates to a following base webpack plugin configuration:
+The `@openshift/dynamic-plugin-sdk-webpack` plugin will output manifest file in the following format:
 
-```JS
-import { container } from 'webpack';
-
-const dynamicPlugin = new container.ModuleFederationPlugin({
-  name: 'backstage.plugin-github-actions',
-  library: {
-    type: 'global',
-    /**
-     * Some library.type has name limitation
-     * for example, if "type": "var" is used, the library.name can contain the "-" character
-    */
-    name: 'backstage.plugin-github-actions'
-  },
-  filename: 'backstage.plugin-github-actions.[contenthash].js',
-  exposes: {
-    PluginRoot: './src/index.ts'
-  },
-  // list of shared modules like "react", "react-dom", etc
-  shared: sharedModules,
-});
+```ts
+type PluginManifest = {
+  name: string;
+  version: string;
+  dependencies?: Record<string, string>;
+  customProperties?: AnyObject;
+  baseURL: string;
+  extensions: Extension[];
```

Comment:
It is the extension definition Vojtech was showing on the call we had. It can be empty and will be probably at least initially. Unless you decide it would be worth to adopt the concept.

Reply:
Alright, could we drop it from the update for now?

Reply:
Sure thing :+1: